pkg/tsdb/cloudwatch/models/request_parser.go

```diff
@@ -66,66 +56,109 @@ func ParseMetricDataQueries(queries []backend.DataQuery, startTime time.Time, en
 			continue
 		}
 
-		if metricsDataQuery.MatchExact == nil {
-			trueBooleanValue := true
-			metricsDataQuery.MatchExact = &trueBooleanValue
+		metricDataQueries[query.RefID] = metricsDataQuery
+	}
+
+	var result []*CloudWatchQuery
+	for refId, mdq := range metricDataQueries {
+		cwQuery := &CloudWatchQuery{
+			Alias:             mdq.Alias,
+			RefId:             refId,
+			Id:                mdq.Id,
+			Region:            mdq.Region,
+			Namespace:         mdq.Namespace,
+			MetricName:        mdq.MetricName,
+			MetricQueryType:   mdq.MetricQueryType,
+			SqlExpression:     mdq.SqlExpression,
+			TimezoneUTCOffset: mdq.TimezoneUTCOffset,
+			Expression:        mdq.Expression,
 		}
 
-		refID := query.RefID
-		cwQuery, err := parseRequestQuery(metricsDataQuery, refID, startTime, endTime)
-		if err != nil {
-			return nil, &QueryError{Err: err, RefID: refID}
+		if err := cwQuery.validateAndSetDefaults(refId, mdq, startTime, endTime); err != nil {
+			return nil, &QueryError{Err: err, RefID: refId}
 		}
+
+		cwQuery.migrateStatisticsAndAlias(mdq, dynamicLabelsEnabled)
+
 		result = append(result, cwQuery)
 	}
 
 	return result, nil
 }
 
-// migrateLegacyQuery is also done in the frontend, so this should only ever be needed for alerting queries
-func migrateLegacyQuery(queries []backend.DataQuery, dynamicLabelsEnabled bool) ([]*backend.DataQuery, error) {
-	migratedQueries := []*backend.DataQuery{}
-	for _, q := range queries {
-		query := q
-		var queryJson *metricsDataQuery
-		err := json.Unmarshal(query.JSON, &queryJson)
-		if err != nil {
-			return nil, err
-		}
+func (q *CloudWatchQuery) migrateStatisticsAndAlias(query metricsDataQuery, dynamicLabelsEnabled bool) {
```

Comment:
Is there a reason to call this `migrateStatisticsAndAlias` instead of keeping it as `migrateLegacyQuery`? In the future I'd imagine we'd add additional migrations here as well.

Reply:
No reason in particular. I had the outer function `migrateLegacyQuery` at some point and put it back under a different name. I've renamed it now.